### PR TITLE
[IMP] event,website_event: ui improvents for online events

### PR DESCRIPTION
**Pull request by adsh-odoo**
Purpose:
- Emphasize that the event is hosted online and doesn't have a physical address.

Specification:
- Add the placeholder for the event on whom the locations are not set. 
-  Add the "online event"  with a localization icon in the front end when no
   venue is set.
- Made the filter for the online event and grouped by the event as per their
   venue.

Task-3384870

### CODE REVIEW

File: addons/event/static/src/js/tests/online_event_groupby_tests.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    id: 1,
+                    name: "My event",
+                    address_id: false,
+                },
+            ],
+        };
+        target = getFixture();
+        setupViewRegistries();
+    });
+    QUnit.test("Test group label for online", async function (assert) {
```
Comment (jeh-odoo) [inline addons/event/static/src/js/tests/online_event_groupby_tests.js:36]: `... for online event in tree view` ?

File: addons/event/static/src/js/tests/online_event_groupby_tests.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        });
+        await openView({
+            res_model: "event.event",
+            views: [[false, "tree"]],
+            context: { group_by: ["address_id"] },
+        });
+
+        assert.strictEqual(target.querySelector(".o_group_name").innerText, "Online (1)");
+    });
+    QUnit.test("Test group label for online event", async function (assert) {
```
Comment (jeh-odoo) [inline addons/event/static/src/js/tests/online_event_groupby_tests.js:55]: `... for online event in kanban view` and you can move the kanban test before the tree one to keep the alphabetical order 🙂

**Review by jeh-odoo — COMMENTED** [review]
Hello @adsh-odoo 
Just some small comments, otherwise LGTM!
Initially, I was wondering if the test was necessary but seems we do it for other apps so we can keep it.
Thanks for the work!
Cheers!

File: addons/event/static/src/views/event_kanban_view.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { kanbanView } from "@web/views/kanban/kanban_view";
+import { KanbanRenderer } from "@web/views/kanban/kanban_renderer";
+
+export class EventKanbanHeader extends KanbanHeader {
+    /**
+     * @override
+     */
+    _getEmptyGroupLabel(fieldName) {
+        if (fieldName === "address_id") {
+            return _t("Online");
```
[the remark this answers is not in the document]
Comment (adsh-odoo) [reply addons/event/static/src/views/event_kanban_view.js:15]: Hello @jeh-odoo 

I tried but not getting `_t ` in `env`. isn't it is removed here https://github.com/odoo/odoo/commit/c07181b20bf4f06b783136ffdb3b7a304be6b136 or am I missing something?

[the remark this answers is not in the document]
Comment (jeh-odoo) [reply addons/event/static/src/views/event_kanban_view.js:15]: Hi! Indeed sorry, you're right. It seems I missed this change and thought we could still do it. You can ignore this comment. Thanks for the info :slightly_smiling_face:

**Comment by adsh-odoo** [comment]
Hello @jeh-odoo :slightly_smiling_face: 

Thanks for the validation.
I have updated the code as per your suggestion. Can you please have a look?

File: addons/website_event/views/event_templates_page_registration.xml [hunk 4]
```diff
@@ -100,6 +100,10 @@
                                     <p class="mb-0"><i class="fa fa-map-marker fa-fw" role="img"/>Get the direction</p>
                                 </a>
                             </div>
+                            <div t-else="" class="d-flex mb-1">
```
Comment (awa-odoo) [inline addons/website_event/views/event_templates_page_registration.xml:103]: Sugg:

```html
<h6 t-else="">
    <i class="fa fa-map-marker" title="Location"/> Online event
</h6>
```

Looks a bit better in terms of alignment and is much more simple.

**Review by awa-odoo — COMMENTED** [review]
Hello @adsh-odoo 

A few comments to try to polish before the merge.

Thanks & cheers!

File: addons/website_event/views/event_templates_list.xml [hunk 5]
```diff
@@ -291,7 +291,18 @@
                             </small>
                         </div>
                         <!-- Location -->
-                        <small class="o_not_editable fw-bold" itemprop="location" t-out="event.address_id" t-options="{'widget': 'contact', 'fields': ['city'], 'no_marker': 'true'}"/>
+                        <t t-if="event.address_id">
+                            <div class="d-flex">
```
[the remark this answers is not in the document]
Comment (adsh-odoo) [reply addons/website_event/views/event_templates_list.xml]: I used the d-flex to put the icon and text horizontally as both are block elements.
By the way, I have fixed the alignment issue.

**Comment by adsh-odoo** [comment]
Hello @awa-odoo :slightly_smiling_face: 
Suggested changes have been done.
Can you please have a look?
Thanks!!

File: addons/event/views/event_event_views.xml [hunk 6]
```diff
@@ -232,9 +232,10 @@
                                         <div class="o_kanban_record_title o_text_overflow" t-att-title="record.name.value">
                                             <field name="name"/>
                                         </div>
-                                        <div t-if="record.address_id.value" class="d-flex ps-1">
+                                        <div class="d-flex ps-1">
                                             <i class="fa fa-map-marker mt-1 me-2 text-center ps-1" title="Location"/>
-                                            <span t-esc="record.address_id.value" class="ms-1"/>
+                                            <span t-if="record.address_id.value" t-out="record.address_id.value" class="ms-1"/>
+                                            <span t-else="" t-out="'Online event'" class="ms-1"/>
```
Comment (awa-odoo) [inline addons/event/views/event_event_views.xml:238]: Please re-read my previous comment, you cannot use t-esc or t-out for translated terms as they will not appear in the POT file, I left the correct suggestion in my previous review.

**Review by awa-odoo — COMMENTED** [review]
Hello @adsh-odoo 

One last thing to fix and we should be good.

Thanks & cheers!

File: addons/event/views/event_event_views.xml (hunk 6 above)
Comment (adsh-odoo) [reply addons/event/views/event_event_views.xml]: Hello @awa-odoo 
My bad I misunderstood that one
Done with the changes. Can you please have a look?

Thanks!!

**Comment by adsh-odoo** [comment]
Hello @awa-odoo 
Can you please have a look :blush: 
Thanks!!

**Pull request closed without merging**